semgrep/semgrep/target_manager.py

```diff
@@ -193,7 +194,12 @@ def match_glob(path: Path, globs: List[str]) -> bool:
         Return true if path or any parent of path matches any glob in globs
         """
         subpaths = [path, *path.parents]
-        return any(p.match(glob) for p in subpaths for glob in globs)
+        return any(
+            wcpathlib.PurePath(p).match(
+                globs, flags=wcpathlib.GLOBSTAR | wcpathlib.DOTGLOB
+            )
+            for p in subpaths
+        )
```

Comment:
wcmatch's `PurePath.match` is slower and looping like this is quite a performance hit.
Let me try another approach.